darts/models/torch_forecasting_model.py

```diff
@@ -327,19 +362,17 @@ def fit_from_dataset(self,
                          verbose: bool = False,
                          epochs: int = 0) -> None:
 
+        raise_if(self.checkpoint_exists and self.force,
```

Comment:
Is there a case where someone would set `force=True` and then would not want to call `reset_model()`. I'm asking because it seems to me that that if someone does set `force=True`, they always want to reset the model, in which case we could automatically call `reset_model()` if `force=True`, instead of requiring the user to do it. But maybe I'm missing something. What do you think?

Reply:
I was doing my review, and before reading your comment @pennfranc I wrote this:

"
Would there be a common use-case for creating a model with `force=True` but then not calling `reset_model()`?
It not really, I would suggest we go ahead and reset the model when `force` is `True`. If users set this to `True` we can take this as a voluntary action IMO.
"

Reply:
Then the entire PR wouldn't make too much sense, because it would make again deleting user's data automatic in some cases. 
How I see it is that we often run notebooks in `Run all` mode. If there is a code that was left with `force=True` that would delete all the data and that might have not been users intention. This little param may get lost in all the code. I case of my approach, user has a chance to either remove `force=True` or do `reset_model()`. With the automatic decision, she may end up with the result she didn't want.
And I understand that users should be careful what they execute, but in the reality they often aren't from different reasons, and this code is covering that case.

Reply:
@pennfranc / @hrzn is this a satisfactory explanation or should we discuss it further?

Reply:
@mkos I'm not very convinced TBH because in any case someone building the model with `force=True` will in all likelihood also be calling `reset_model()`. So it will be equally likely to forget this piece of code as well in a notebook and have the same problem when launching "run all". 

But more in general, I'm not convinced that we should proactively try to anticipate all the sort of programming and environment issues users might have; at least when it comes at the expense of clarity (as is the case here - it's confusing to have a `force` parameter that doesn't reset the model). I think having a clear & straightforward API is the best way to avoid errors. Of course some users might mess up their execution runs in the notebooks, but frankly this sort of issues happening with notebooks is a bit out of our realm I think. We have to assume users will use notebooks correctly, and provide the clearest possible API.

Reply:
I think the part we should keep is the overriding of `epochs` parameter in `fit()` when provided. Also we can keep the `reset_model()` method, which could be convenient in some cases. WDYT?

Reply:
so to sum up:
1. If `force=False` and checkpoints exist - raise exception
2. If `force=True` and checkpoints exist - run reset_model() quietly
3. fit does not emit any warning.
correct?

Reply:
That sounds good to me

Reply:
Me too 👍